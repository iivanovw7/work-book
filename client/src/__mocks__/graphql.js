import {
  GET_POST, GET_POSTS, GET_TAGS, GET_USER, FIND_POSTS_BY_TAG, FIND_POSTS_BY_KEYWORD
} from '../graphql/queries';
import { DELETE_POST, ADD_POST, UPDATE_POST } from '../graphql/mutations';

export const gqlMocks = [
  /**
	 * [0] Returned after success GET POST request
	 */
  {
    request: {
      query: GET_POST,
      variables: {
        _id: '1'
      }
    },
    result: {
      data: {
        getPost: {
          _id: '1',
          text: 'Post text',
          title: 'Post title',
          subject: 'Post subject',
          created: 'Post date',
          author: 'Post author',
          tags: ['1', '2', '3']
        }
      }
    }
  },
  /**
	 * [1] Returned after errored GET POST request
	 */
  {
    request: {
      query: GET_POST,
      variables: {
        _id: '1'
      }
    },
    error: new Error('aw shucks')
  },
  /**
	 * [2] Returned after success DELETE POST mutation
	 */
  {
    request: {
      query: DELETE_POST,
      variables: {
        _id: '1'
      }
    },
    result: {
      deletePost: {
        text: 'Post text',
        title: 'Post title',
        subject: 'Post subject',
        tags: ['1', '2', '3']
      }
    }
  },
  /**
	 * [3] Returned after errored DELETE POST mutation
	 */
  {
    request: {
      query: DELETE_POST,
      variables: {
        _id: '1'
      }
    },
    error: new Error('aw shucks')
  },
  /**
	 * [4] Returned after success ADD POST mutation
	 */
  {
    request: {
      query: ADD_POST,
      variables: {
        text: 'Post text',
        title: 'Post title',
        subject: 'Post subject',
        tags: ['1', '2', '3']
      }
    },
    result: {
      addPost: {
        text: 'Post text',
        title: 'Post title',
        subject: 'Post subject',
        tags: ['1', '2', '3']
      }
    }
  },
  /**
	 * [5] Returned after errored ADD POST mutation
	 */
  {
    request: {
      query: ADD_POST,
      variables: {
        text: 'Post text',
        title: 'Post title',
        subject: 'Post subject',
        tags: ['1', '2', '3']
      }
    },
    error: new Error('aw shucks')
  },
  /**
	 * [6] Returned after success UPDATE POST mutation
	 */
  {
    request: {
      query: UPDATE_POST,
      variables: {
        _id: '1',
        text: 'Post text',
        title: 'Post title',
        subject: 'Post subject',
        tags: ['1', '2', '3']
      }
    },
    result: {
      data: {
        getPost: {
          _id: '1',
          text: 'Post text',
          title: 'Post title',
          subject: 'Post subject',
          created: 'Post date',
          author: 'Post author',
          tags: ['1', '2', '3']
        }
      }
    }
  },
  /**
	 * [7] Returned after success UPDATE POST mutation
	 */
  {
    request: {
      query: UPDATE_POST,
      variables: {
        _id: '1',
        text: 'Post text',
        title: 'Post title',
        subject: 'Post subject',
        tags: ['1', '2', '3']
      }
    },
    error: new Error('aw shucks')
  },
  /**
	 * [8] Returned after success GET POSTS query
	 */
  {
    request: {
      query: GET_POSTS
    },
    result: {
      data: {
        getPosts: [
          {
            _id: '5d283694ddc2cb244f84bdca',
            author: '5cbeba32fb6fc0265f273c94',
            created: '1562915551933',
            subject: 'Cross browser solution for handling mouse wheel events in Angular with Rx.js for',
            tags: ['1', '2', '3'],
            text: 'Post text',
            title: 'Custom Angular mouseWheel scroll directive'
          }
        ]
      }
    }
  },
  /**
	 * [9] Returned after error after GET POSTS query
	 */
  {
    request: {
      query: GET_POSTS
    },
    error: new Error('aw shucks')
  },
  /**
	 * [10] Returned after success GET TAGS query
	 */
  {
    request: {
      query: GET_TAGS
    },
    result: {
      data: {
        getTags: ['1', '2', '3']
      }
    }
  },
  /**
	 * [11] Returned after errored GET TAGS query
	 */
  {
    request: {
      query: GET_TAGS
    },
    error: new Error('aw shucks')
  },
  /**
	 *  [12] Returned after success GET USER query
	 */
  {
    request: {
      query: GET_USER,
      variables: {
        _id: '1'
      }
    },
    result: {
      data: {
        getUser: {
          _id: '1',
          email: 'Email',
          phone: 'Phone',
          name: 'Name',
          surname: 'Surname',
          created: 'Date',
          role: 'Role'
        }
      }
    }
  },
  /**
	 *  [13] Returned after errored GET USER query
	 */
  {
    request: {
      query: GET_USER,
      variables: {
        _id: '1'
      }
    },
    error: new Error('aw shucks')
  },
  /**
   *  [14] Returned after FIND_POSTS_BY_TAG query
   */
  {
    request: {
      query: FIND_POSTS_BY_TAG,
      variables: {
        tag: '1'
      }
    },
    result: {
      data: {
        findPostsByTag: [
          {
            _id: '5d283694ddc2cb244f84bdca',
            author: '5cbeba32fb6fc0265f273c94',
            created: '1562915551933',
            subject: 'Cross browser solution for handling mouse wheel events in Angular with Rx.js for',
            tags: ['1', '2', '3'],
            text: 'Post text',
            title: 'Custom Angular mouseWheel scroll directive'
          }
        ]
      }
    }
  },
  /**
   * [15] Returned after errored FIND_POSTS_BY_TAG query
   */
  {
    request: {
      query: FIND_POSTS_BY_TAG,
      variables: {
        tag: '1'
      }
    },
    error: new Error('aw shucks')
  },
  /**
   *  [16] Returned after FIND_POSTS_BY_KEYWORD query
   */
  {
    request: {
      query: FIND_POSTS_BY_KEYWORD,
      variables: {
        keyword: 'Custom'
      }
    },
    result: {
      data: {
        findPostsByKeyword: [
          {
            _id: '5d283694ddc2cb244f84bdca',
            author: '5cbeba32fb6fc0265f273c94',
            created: '1562915551933',
            subject: 'Cross browser solution for handling mouse wheel events in Angular with Rx.js for',
            tags: ['1', '2', '3'],
            text: 'Post text',
            title: 'Custom Angular mouseWheel scroll directive'
          }
        ]
      }
    }
  },
  /**
   * [17] Returned after errored FIND_POSTS_BY_KEYWORD query
   */
  {
    request: {
      query: FIND_POSTS_BY_KEYWORD,
      variables: {
        keyword: 'Custom'
      }
    },
    error: new Error('aw shucks')
  }
];
