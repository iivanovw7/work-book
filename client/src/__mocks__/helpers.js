import * as constants from '../constants';

export const convertTimestampOptions = [
  [1305593400, constants.TIMESTAMP_UNITS.MS, '{YYYY} MM-DD SSS [Z] A'],
  [1305593400, constants.TIMESTAMP_UNITS.SEC, '{YYYY} MM-DD SSS [Z] A'],
  [undefined, constants.TIMESTAMP_UNITS.MS, 'MMMM DD, YYYY'],
  [null, 'units', null]
];

export const convertTimestampExpected = [
  '{1970} 01-16 000 Z AM',
  '{2011} 05-17 000 Z AM',
  'Invalid Date',
  'Invalid Date'
];
