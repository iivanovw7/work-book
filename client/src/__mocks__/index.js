export * from './loginResponse';
export * from './localizedText';
export * from './graphql';
export * from './helpers';
