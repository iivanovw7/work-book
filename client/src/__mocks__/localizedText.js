import * as localized from '../assets/locales.json';

// Returns localised text for UI elements
export const localizedText = {
  locales: localized.locales,
  login: localized.login,
  navigation: localized.navigation,
  pageNotFound: localized.pageNotFound
};
