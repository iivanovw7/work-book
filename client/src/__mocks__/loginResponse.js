export const loginResponse = {
  // eslint-disable-next-line max-len
  jwt: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ZDA1ZmY0YTMyM2ZjNDZjMmE4YWMxNzUiLCJpZCI6IjVkMDVmZjRhMzIzZmM0NmMyYThhYzE3NSIsImlhdCI6MTU2MTAzMTk4MiwiZXhwIjoxNTYzNjIzOTgyfQ.ldBlLabhjPJg3-TwvuE5MwfMnhLRk7lbBm22KNfV4ws',
  user: {
    _id: '5d05ff4a323fc46c2a8ac175',
    confirmed: true,
    blocked: false,
    username: 'manager',
    email: '9iiiuk3@gmail.com',
    provider: 'local',
    createdAt: '2019-06-16T08:35:22.613Z',
    updatedAt: '2019-06-16T08:35:22.771Z',
    __v: 0,
    id: '5d05ff4a323fc46c2a8ac175',
    role: {
      _id: '5d052d6f0f0d3e3186cd34a4',
      name: 'Authenticated',
      description: 'Default role given to authenticated user.',
      type: 'authenticated',
      __v: 0
    }
  }
};
