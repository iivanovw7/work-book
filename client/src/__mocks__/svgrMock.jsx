import React from 'react';

export default 'SvgrURL';
const SvgrMock = React.forwardRef((props, ref) => <icon-mock ref={ref} {...props} />);

export const ReactComponent = SvgrMock;

//module.exports = { ReactComponent: 'icon-mock' };
