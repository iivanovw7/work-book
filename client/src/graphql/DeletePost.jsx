import PropTypes from 'prop-types';
import React from 'react';
import { Mutation } from 'react-apollo';
import { Redirect, withRouter } from 'react-router-dom';
import styled, { ThemeProvider } from 'styled-components';
import Button from '../elements/UI/Button';
import Spinner from '../elements/UI/Spinner';
import { DELETE_POST } from './mutations';
import { GET_POSTS } from './queries';
/* eslint no-underscore-dangle: 0 */

const StyledContainer = styled.div`
	float: right;
	max-height: 60px;
`;

const DeletePost = (props) => {
  const {
    post, text, history, theme, locale
  } = props;

  return (
    <Mutation
      mutation={DELETE_POST}
      refetchQueries={() => [{ query: GET_POSTS }]}
    >
      {(deletePost, { loading, error, data }) => (
        <StyledContainer>
          <Button
            variant="alert"
            text={text.navigation.delete[locale]}
            theme={theme}
            maxWidth="7em"
            handleClick={() => {
              deletePost({
                variables: {
                  _id: post._id
                }
              });
              return history.push('/posts');
            }}
          />
        </StyledContainer>
      )}
    </Mutation>
  );
};

export default withRouter(DeletePost);

DeletePost.propTypes = {
  history: PropTypes.object.isRequired,
  post: PropTypes.object.isRequired,
  text: PropTypes.object.isRequired,
  theme: PropTypes.string.isRequired,
  locale: PropTypes.string.isRequired
};
