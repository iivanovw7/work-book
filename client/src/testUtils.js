import { createMemoryHistory } from 'history';
import * as localized from './assets/locales.json';

// Mock browser history
export const history = createMemoryHistory('/dashboard');

// Grid config object for testing
export const mappedSizes = {
	xs: 12,
	sm: 12,
	md: 12,
	lg: 11
};

export const localizedText = {
	locales: localized.locales,
	topBar: localized.topBar,
	login: localized.login,
	navigation: localized.navigation,
	pageNotFound: localized.pageNotFound
};
