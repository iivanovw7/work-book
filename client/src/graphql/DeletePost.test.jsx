import React from 'react';
import wait from 'waait';
import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from 'react-apollo/test-utils';
import * as mocks from '../__mocks__';
import * as testUtils from '../testUtils';
import DeletePost from './DeletePost';

jest.mock('../config/apiURL', () => () => '/api');
jest.mock('storeon/react', () => () => ({
  locale: 'eng',
  theme: 'dark',
  search: ''
}));
jest.mock('../elements/UI/Button', () => () => (
  <div id="Button" />
));


describe('Testing DeletePost() graphql query: ', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    locale: 'eng',
    text: testUtils.localizedText,
    theme: 'dark',
    history: testUtils.history,
    post: mocks.gqlMocks[0].result.data.getPost
  };

  const Composition = () => (
    <MemoryRouter>
      <MockedProvider mocks={[mocks.gqlMocks[0]]} addTypename={false} removeTypename>
        <DeletePost _id={1} id="1" {...props} />
      </MockedProvider>
    </MemoryRouter>
  );

  it('Should render Component and match snapshot', async () => {
    const component = testUtils.suppressConsoleWarnings(
      <Composition />,
      'mount',
      console
    );


    await wait(0);
    await component.update();
    expect(component).toMatchSnapshot();
  });

  it('Should render Component with correct data', async () => {
    const component = testUtils.suppressConsoleWarnings(
      <Composition />,
      'mount',
      console
    );

    await wait(0);
    await component.update();
    expect(component.find('#Button').length).toEqual(1);
    expect(component.find({ text: 'Delete' }).length).toEqual(1);
  });
});
