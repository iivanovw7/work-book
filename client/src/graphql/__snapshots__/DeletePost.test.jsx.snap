// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing [[33mDeletePost[39m] graphql query:  Should render Component and match snapshot 1`] = `ReactWrapper {}`;
